lib/python/rose/suite_clean.py

```diff
@@ -102,11 +102,37 @@ def _clean(self, suite_name, only_items=None):
                     if item:
                         loc_rel = os.path.join(suite_dir_rel, item)
                     locs.append(os.path.join(item_root, loc_rel))
+                    roots.add(item_root)
             if self.host_selector.is_local_host(auth):
+                # Clean relevant items
                 for loc in locs:
                     loc = os.path.abspath(env_var_process(loc))
                     for name in sorted(glob(loc)):
                         engine.fs_util.delete(name)
+                # Clean empty directories
+                # Change directory to root level to avoid cleaning them as well
+                # For cylc suites, e.g. it can clean up to an empty "cylc-run/"
+                # directory.
+                for root in sorted(roots):
+                    cwd = os.getcwd()
+                    if root:
+                        try:
+                            os.chdir(env_var_process(root))
+                        except OSError:
+                            continue
+                    # Reverse sort to ensure that e.g. "share/cycle/" is
+                    # cleaned before "share/"
+                    for name in sorted(self.CLEANABLE_PATHS, reverse=True):
```

Comment:
> Any future additions/changes to `CLEANABLE_PATHS` could affect the sorting or have further consequences for the order in which things must be done not necessarily capture-able by a sort operation

The logic here already has a comment, and the `sorted` logic should future-proof any unordered addition to `CLEANABLE_PATHS`. What kind of issues do you have in mind that are not capture-able by a sort?